modules/account-lib/src/coin/algo/keyPair.ts

```diff
@@ -1,30 +1,58 @@
+/* eslint-disable no-unused-vars */
+import algosdk, { encodeAddress } from 'algosdk';
 import { Ed25519KeyPair } from '../baseCoin';
-import { AddressFormat } from '../baseCoin/enum';
-import { NotImplementedError } from '../baseCoin/errors';
 import { DefaultKeys, KeyPairOptions } from '../baseCoin/iface';
+import utils from './utils';
 
 export class KeyPair extends Ed25519KeyPair {
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param { KeyPairOptions } source Either a master seed, a private key, or a public key
+   */
   constructor(source?: KeyPairOptions) {
     super(source);
   }
 
   /** @inheritdoc */
   recordKeysFromPrivateKeyInProtocolFormat(prv: string): DefaultKeys {
-    throw new NotImplementedError('recordKeysFromPrivateKeyInProtocolFormat not implemented');
+    const decodedSeed = utils.decodeSeed(prv);
+    const hexSecretKey = Buffer.from(decodedSeed.seed);
+
+    return utils.keyPairFromSeed(hexSecretKey).keyPair;
   }
 
   /** @inheritdoc */
-  recordKeysFromPublicKeyInProtocolFormat(prv: string): DefaultKeys {
-    throw new NotImplementedError('recordKeysFromPublicKeyInProtocolFormat not implemented');
+  recordKeysFromPublicKeyInProtocolFormat(pub: string): DefaultKeys {
+    const publicKey = algosdk.decodeAddress(pub).publicKey;
+    return { pub: utils.toHex(publicKey) };
   }
 
   /** @inheritdoc */
-  getAddress(format: AddressFormat): string {
-    throw new NotImplementedError('getAddress not implemented');
+  getAddress(): string {
+    return encodeAddress(utils.hexStringToUInt8Array(this.keyPair.pub));
   }
 
   /** @inheritdoc */
-  getKeys(): void {
-    throw new NotImplementedError('getKeys not implemented');
+  getKeys(): DefaultKeys {
+    // TODO(https://bitgoinc.atlassian.net/browse/STLX-6062): refactor this method
+    // should return the pub and prv in the Algorand encoded format
+    const result: DefaultKeys = { pub: this.keyPair.pub };
+
+    if (this.keyPair.prv) {
+      result.prv = this.keyPair.prv;
+    }
+
+    return result;
+  }
+
+  /**
+   * Getting the base64 private key for signing an algo transaction.
+   *
+   * @returns {Uint8Array} base64 private key
+   * @see https://developer.algorand.org/docs/features/accounts/#transformation-private-key-to-base64-private-key
+   */
+  getSigningKey(): Uint8Array {
+    return utils.toUint8Array(this.keyPair.prv + this.keyPair.pub);
```

Comment:
Do these need to be converted to their byte representations? What if you don't have the prv? Is that case worth checking here?

Reply:
If you haven't the PrivateKey, you shouldn't be able to sign the transaction. That byte representation is the secretKey, that is required to sign on Algorand.

Reply:
Also are there any tests for this function?

Reply:
That have sense. I'll add it and I'll make sure that are tests related to that.